workflow/templateresolution/context.go

```diff
@@ -79,22 +105,41 @@ func (ctx *Context) GetTemplateByName(name string) (*wfv1.Template, error) {
 	return tmpl.DeepCopy(), nil
 }
 
+func (ctx *Context) GetTemplateGetterFromRef(tmplRef *wfv1.TemplateRef) (wfv1.TemplateGetter, error) {
+	if tmplRef.ClusterScope {
+		return ctx.cwftmplGetter.Get(tmplRef.Name)
+	}
+	return ctx.wftmplGetter.Get(tmplRef.Name)
+}
+
 // GetTemplateFromRef returns a template found by a given template ref.
 func (ctx *Context) GetTemplateFromRef(tmplRef *wfv1.TemplateRef) (*wfv1.Template, error) {
 	ctx.log.Debug("Getting the template from ref")
-
-	wftmpl, err := ctx.wftmplGetter.Get(tmplRef.Name)
-	if err != nil {
-		if apierr.IsNotFound(err) {
-			return nil, errors.Errorf(errors.CodeNotFound, "workflow template %s not found", tmplRef.Name)
+	var template *wfv1.Template
+	if tmplRef.ClusterScope {
+		cwftmpl, err := ctx.cwftmplGetter.Get(tmplRef.Name)
```

Comment:
This code can be refactored as both `cwftmpl` and `wftmpl` implement `WorkflowTemplateInterface`. Or, just use `GetTemplateGetterFromRef` to get `WorkflowTemplateInterface` because both `cwftmplGetter` and  `wftmplGetter ` implement `TemplateGetter`.